src/elements/content-sidebar/versions/__tests__/VersionsItem.test.js

```diff
@@ -238,5 +238,111 @@ describe('elements/content-sidebar/versions/VersionsItem', () => {
+            expect(actions.prop('showPromote')).toBe(false);
+            expect(actions.prop('showPreview')).toBe(permissions.can_preview);
+            expect(wrapper.find(ReadableTime)).toBeTruthy();
+        });
+
+        test.each`
+            permissions
+            ${{ can_delete: true, can_download: false, can_preview: false, can_upload: true }}
+            ${{ can_delete: true, can_download: false, can_preview: false, can_upload: false }}
+            ${{ can_delete: false, can_download: false, can_preview: false, can_upload: true }}
+            ${{ can_delete: false, can_download: false, can_preview: false, can_upload: false }}
+        `('should not render actions based on permissions when is archive file', ({ permissions }) => {
+            const wrapper = getWrapper({
+                version: getVersion({ permissions }),
+                isArchiveFile: true,
+            });
+            const actions = wrapper.exists(VersionsItemActions);
+            const button = wrapper.find(VersionsItemButton);
+
+            expect(button.prop('isDisabled')).toBe(!permissions.can_preview);
+            expect(actions).toBeFalsy();
+            expect(wrapper.find(ReadableTime)).toBeTruthy();
+        });
+
+        test.each`
+            action       | showDelete | showDownload | showPreview | showPromote | showRestore
+            ${'restore'} | ${false}   | ${true}      | ${true}     | ${false}    | ${false}
+            ${'upload'}  | ${false}   | ${true}      | ${true}     | ${false}    | ${false}
+        `(
+            "should show actions correctly when the version's action is $action and is archive file",
+            ({ action, showDelete, showDownload, showPreview, showPromote, showRestore }) => {
+                selectors.getVersionAction.mockReturnValueOnce(action);
+
+                const wrapper = getWrapper({
+                    version: getVersion({
+                        permissions: {
+                            can_delete: true,
+                            can_download: true,
+                            can_preview: true,
+                            can_upload: true,
+                        },
+                    }),
+                    isArchiveFile: true,
+                });
+                const actions = wrapper.find(VersionsItemActions);
+                const button = wrapper.find(VersionsItemButton);
+
+                expect(button.prop('isDisabled')).toBe(!showPreview);
+                expect(actions.prop('showDelete')).toBe(showDelete);
+                expect(actions.prop('showDownload')).toBe(showDownload);
+                expect(actions.prop('showPromote')).toBe(showPromote);
+                expect(actions.prop('showPreview')).toBe(showPreview);
+                expect(actions.prop('showRestore')).toBe(showRestore);
+                expect(wrapper.find(ReadableTime)).toBeTruthy();
+                expect(wrapper).toMatchSnapshot();
```

Comment:
We generally want to avoid adding new snapshot tests, especially for larger components. Let's rely on explicit assertions, instead.

Reply:
I've moved all the tests for this component to react-testing-library, so I removed snapshots completely there